commander.ts

```diff
@@ -185,11 +214,17 @@ export class Commander implements CmdVisitor<Value | null> {
     });
   }
 
-  async evalProgram(program: Program, filename: string): Promise<void> {
+  async evalProgram(
+    [program, parseWarning]: ParseResult<Program>,
+    filename: string,
+  ): Promise<void> {
     return tracer.span('evalProgram', async () => {
       let chunk: Chunk;
+      let compilerWarning: ParseWarning | null = null;
       try {
-        chunk = compile(program, { filename });
+        const result = compile(program, { filename });
+        chunk = result[0];
+        compilerWarning = result[1];
       } catch (err) {
         if (err instanceof Exception) {
```

Comment:
I'd need to catch `ParseError` and merge `parseWarning` with it before throwing here as well (and in similar spots).